### PR TITLE
feat(team-workflow): Add team to GroupSubscription

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr adds back the addition of team as a field on GroupSubscription (originally from https://github.com/getsentry/sentry/pull/55805 , removed in https://github.com/getsentry/sentry/pull/55870) .

### CODE REVIEW

**Comment by roggenkemper** [comment]
don't merge until ops has run the migration

**Review by markstory — APPROVED** [review]
Migration looks good to me. We'll need to wait for schema changes to be applied in saas before merging this though.

**Pull request merged**
